src/com/facebook/buck/maven/Resolver.java

```diff
@@ -92,73 +114,133 @@
   private final Path buckRepoRoot;
   private final Path buckThirdPartyRelativePath;
   private final LocalRepository localRepo;
-  private final ImmutableList<RemoteRepository> repos;
   private final ServiceLocator locator;
   private final VersionScheme versionScheme = new GenericVersionScheme();
+  private final ModelBuilder modelBuilder;
+
+  private ImmutableList<Dependency> initialRequest;
+  private ImmutableList<RemoteRepository> repos;
 
   public Resolver(
       Path buckRepoRoot,
       Path relativeThirdParty,
-      Path localRepoPath,
-      String... repoUrls) {
+      Path localRepoPath) {
     this.buckRepoRoot = buckRepoRoot;
     this.buckThirdPartyRelativePath = relativeThirdParty;
     this.localRepo = new LocalRepository(localRepoPath.toFile());
+    this.initialRequest = ImmutableList.of();
+    this.repos = ImmutableList.of();
 
+    this.modelBuilder = new DefaultModelBuilderFactory().newInstance()
+        .setProfileSelector(new DefaultProfileSelector())
+        .setPluginConfigurationExpander(new DefaultPluginConfigurationExpander())
+        .setPluginManagementInjector(new DefaultPluginManagementInjector())
+        .setDependencyManagementImporter(new DefaultDependencyManagementImporter())
+        .setDependencyManagementInjector(new DefaultDependencyManagementInjector());
+
+    this.locator = AetherUtil.initServiceLocator();
+  }
+
+  public void addRepositories(String... repoUrls) {
     ImmutableList.Builder<RemoteRepository> builder = ImmutableList.builder();
+    builder.addAll(repos);
     for (int i = 0; i < repoUrls.length; i++) {
       builder.add(AetherUtil.toRemoteRepository(repoUrls[i]));
     }
     this.repos = builder.build();
+  }
 
-    this.locator = AetherUtil.initServiceLocator();
+  public void addArtifacts(String... mavenCoords) {
+    initialRequest = ImmutableList.<Dependency>builder()
+        .addAll(initialRequest)
+        .addAll(getArtifactsFromString(Arrays.asList(mavenCoords)))
+        .build();
+  }
+
+  public void loadPomFile(Path pomFile) {
+    Model model = loadPomModel(pomFile);
+    initialRequest = ImmutableList.<Dependency>builder()
+        .addAll(initialRequest)
+        .addAll(getArtifactsFromPom(model))
+        .build();
+    repos = ImmutableList.<RemoteRepository>builder()
+        .addAll(repos)
+        .addAll(getReposFromPom(model))
+        .build();
   }
 
-  public void resolve(String... mavenCoords) throws RepositoryException, IOException {
-    RepositorySystem repoSys = locator.getService(RepositorySystem.class);
-    RepositorySystemSession session = newSession(repoSys);
+  public void resolve() throws Exception {
```

Comment:
Also, this method looks quite big. Maybe splitting it to smaller ones can help a bit.